A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/LogicElements/A32NX_FlightPlanManager.js

```diff
@@ -395,20 +396,18 @@ class FlightPlanManager {
     }
 
     _incrementFlightPlanVersion() {
-        // Get most up to date version in case updateFlightPlan hasn't been called yet before the last increment.
-        const currentVersion = SimVar.GetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number');
-        SimVar.SetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number', currentVersion + 1);
+        this._newFlightPlanVersion++;
+        SimVar.SetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number', this._newFlightPlanVersion);
     }
 
     updateFlightPlan(callback = () => { }, log = false) {
-        const newVersion = SimVar.GetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number');
-        if (newVersion === this._currentFlightPlanVersion) {
+        if (this._newFlightPlanVersion === this._currentFlightPlanVersion) {
```

Comment:
you also need to check the sim var value here, as this is the way FlightPlanManager from MCDU and ND communicate with each other - there are 3 instances: MCDU has one, Caption ND has one, and FO ND has one.

Reply:
Fixed, thanks!